### ISSUE TITLE
Ignore DDL based on regex.

### ISSUE THREAD

**Issue by subkanthi** [Altinity/clickhouse-sink-connector#895]
Add functionality so that the DDL regex defined in a configuration variable will be ignored and not executed.

**Comment by subkanthi** [comment]
Add to `config.yml`
```
#ignore.ddl.regex: If set, the connector will ignore DDL events that match the regex.
ignore.ddl.regex: "(?i)(ANALYZE PARTITION).*"

```

Tested:
MySQL:
```
CREATE TABLE sales (     id INT NOT NULL,     sale_date DATE NOT NULL,     amount DECIMAL(10, 2),     PRIMARY KEY (id, sale_date) ) PARTITION BY RANGE (YEAR(sale_date)) (     PARTITION p2020 VALUES LESS THAN (2021),     PARTITION p2021 VALUES LESS THAN (2022),     PARTITION p2022 VALUES LESS THAN (2023),     PARTITION pfuture VALUES LESS THAN MAXVALUE );

```
```
alter table sales analyze partition p2022;
```

```
clickhouse-sink-connector-lt_1  | 2024-11-05 02:02:32.685 INFO  - Ignoring DDL: alter table sales analyze partition p2022 as it matches the regex: (?i)(ANALYZE PARTITION).*
clickhouse-sink-connector-lt_1  | 2024-11-05 02:02:32.685 INFO  - Ignored Source DB DDL: alter table sales analyze partition p2022 Snapshot:false
```

**Comment by subkanthi** [comment]
Added fix in #916, if the `ignore.ddl.regex` is not defined, sink connector should not throw an error.
